### PR TITLE
Gal 703/setup use of nextauth library

### PR DESCRIPTION
**Pull request by Amar-NHS**
## Description

Added NextAuth Library for Authentication, setup basic configuration to authenticate with Credentials for now. Plan is to build on top of this to add CIS2 Identity authentication and authorisation mechanism along with other ACs to meet the requirements later.

## Type of changes

- [ ] Refactoring (non-breaking change)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would change existing functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)

## Checklist

- [ ] I am familiar with the [contributing guidelines](../docs/CONTRIBUTING.md)
- [x] I have followed the code style of the project
- [ ] I have added tests to cover my changes
- [ ] I have updated the documentation accordingly
- [ ] This PR is a result of pair or mob programming

---

## Sensitive Information Declaration

To ensure the utmost confidentiality and protect your and others privacy, we kindly ask you to NOT including [PII (Personal Identifiable Information) / PID (Personal Identifiable Data)](https://digital.nhs.uk/data-and-information/keeping-data-safe-and-benefitting-the-public) or any other sensitive data in this PR (Pull Request) and the codebase changes. We will remove any PR that do contain any sensitive information. We really appreciate your cooperation in this matter.

- [ ] I confirm that neither PII/PID nor sensitive data are included in this PR and the codebase changes.

### CODE REVIEW

File: src/app/api/auth/[...nextauth]/route.ts [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+      // You can pass any HTML attribute to the <input> tag through the object.
+      credentials: {
+        email: { label: "Email", type: "text", placeholder: "Enter Email" },
+        password: { label: "Password", type: "password" },
+      },
+      async authorize(credentials, req) {
+        if (!credentials || !credentials.email || !credentials.password) {
+          return null;
+        }
+        console.log(users.find((item) => item.email === "amar.rai5@nhs.net"));
```
Comment (chenglawson1) [inline src/app/api/auth/[...nextauth]/route.ts:22]: Is this console log with your email needed?

File: src/app/api/auth/[...nextauth]/route.ts [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+    // Choose how you want to save the user session.
+    // The default is `"jwt"`, an encrypted JWT (JWE) stored in the session cookie.
+    // If you use an `adapter` however, we default it to `"database"` instead.
+    // You can still force a JWT session by explicitly defining `"jwt"`.
+    // When using `"database"`, the session cookie will only contain a `sessionToken` value,
+    // which is used to look up the session in the database.
+    // strategy: "database",
+
+    // Seconds - How long until an idle session expires and is no longer valid.
+    maxAge: 60 * 10, // 10 min
```
Comment (chenglawson1) [inline src/app/api/auth/[...nextauth]/route.ts:75]: I think the requirement is 15 minutes, see GAL-358.

Comment (Amar-NHS) [reply src/app/api/auth/[...nextauth]/route.ts]: thanks for spotting that. its set to 15 mins as requirement.

File: src/app/context/AutoSignOutProvider.jsx
Comment (avi312NHS) [inline src/app/context/AutoSignOutProvider.jsx:1]: happy to look over this one tomorrow once your changes are merged in and integrate our work

**Review by avi312NHS — APPROVED** [review]

**Review by chenglawson1 — APPROVED** [review]

**Review by NeerjaGoyalNHS — APPROVED** [review]
LGTM

**Pull request merged**